app/scripts/controllers/serviceInstances.js

```diff
@@ -0,0 +1,98 @@
+'use strict';
+
+angular.module('openshiftConsole')
+  .controller('ServiceInstancesController', function ($scope,
+                                                      $filter,
+                                                      $routeParams,
+                                                      APIService,
+                                                      BindingService,
+                                                      Constants,
+                                                      DataService,
+                                                      LabelFilter,
+                                                      Logger,
+                                                      ProjectsService) {
+    $scope.alerts = {};
+    $scope.bindingsByInstanceRef = {};
+    $scope.emptyMessage = "Loading...";
+    $scope.labelSuggestions = {};
+    $scope.projectName = $routeParams.project;
+    $scope.serviceClasses = {};
+    $scope.serviceInstances = {};
+    $scope.unfilteredServiceInstances = {};
+
+    var watches = [];
+
+    var updateFilter = function() {
+      $scope.serviceInstances = LabelFilter.getLabelSelector().select($scope.unfilteredServiceInstances);
+    };
+
+    var sortServiceInstances = function() {
+      $scope.unfilteredServiceInstances = BindingService.sortServiceInstances($scope.unfilteredServiceInstances, $scope.serviceClasses);
+    };
+
+    ProjectsService
+      .get($routeParams.project)
+      .then(_.spread(function(project, context) {
+        $scope.project = project;
+        $scope.projectContext = context;
+
+        watches.push(DataService.watch({
+          group: 'servicecatalog.k8s.io',
+          resource: 'bindings'
+        }, context, function(bindings) {
+          var bindingsByName = bindings.by('metadata.name');
+          $scope.bindingsByInstanceRef = _.groupBy(bindingsByName, 'spec.instanceRef.name');
+        }));
+
+        watches.push(DataService.watch({
+          group: 'servicecatalog.k8s.io',
+          resource: 'instances'
```

Comment:
This will need to change to `'serviceinstances'` when https://github.com/openshift/origin/pull/16150 is merged.